src/cloud-element-templates/Linter.js

```diff
@@ -0,0 +1,116 @@
+/**
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership.
+ *
+ * Camunda licenses this file to you under the MIT; you may not use this file
+ * except in compliance with the MIT License.
+ */
+
+import StaticResolver from 'bpmnlint/lib/resolver/static-resolver';
+import ElementTemplates from './ElementTemplates';
+import { getPropertyValue, validateProperty } from './util/propertyUtil';
+
+import { applyConditions } from './Condition';
+
+import BpmnModdle from 'bpmn-moddle';
+import zeebeModdle from 'zeebe-bpmn-moddle/resources/zeebe';
+
+import { Validator } from './Validator';
+
+export const elementTemplateLintRule = ({ templates = [] }) => {
+  const moddle = new BpmnModdle({ zeebe: zeebeModdle });
+
+  const validator = new Validator(moddle).addAll(templates);
+  const validTemplates = validator.getValidTemplates();
+
+  // We use the ElementTemplates Module without the required bpmn-js modules
+  // As we only use it to facilitate template ID and version lookup,
+  // access to commandstack etc. is not required
+  const elementTemplates = new ElementTemplates();
+  elementTemplates.set(validTemplates);
+
+  function check(node, reporter) {
+
+    let template = elementTemplates.get(node);
+
+    const templateId = elementTemplates._getTemplateId(node);
+
+    // Handle missing template
+    if (templateId && !template) {
+      reporter.report(
+        node.id,
+        'Linked element template not found',
+        {
+          name: node.name
+        }
+      );
+      return;
+    }
+
+    if (!template) {
+      return;
+    }
+
+    template = applyConditions(node, template);
+
+    // Check attributes
+    template.properties.forEach((property) => {
+      const value = getPropertyValue(node, property);
+      const error = validateProperty(value, property);
+
+      if (!error) {
+        return;
+      }
+
+      reporter.report(
+        node.id,
+        error,
+        {
+          propertiesPanel: {
+            entryIds: [ getEntryId(property, template) ]
+          },
+          name: node.name
+        }
+      );
+    });
+  }
+
+  return {
+    check
+  };
+
+};
+
+
+export const ElementTemplateLinterPlugin = function(templates) {
```

Comment:
It took me some time to understand what's going on here. So basically we have a bpmnlint-plugin buried in the the bpmn-js-element-templates project that is hidden in a file called `Linter` which isn't a linter but a plugin. Maybe we can at least rename the file so it's less confusing.

Reply:
Agreed, let's make the name clearer --> [ba1d96e](https://github.com/bpmn-io/bpmn-js-element-templates/pull/3/commits/ba1d96ea1936be4ae19ef82b9598547c0e16237d)

The rule is only exported for the testing and not exposed in the final bundle. There, only the Plugin is exported: https://github.com/bpmn-io/bpmn-js-element-templates/blob/ba1d96ea1936be4ae19ef82b9598547c0e16237d/src/index.js#L6

Reply:
✔️